In [1]:
data1=read.table("clipboard",header=T)
head(data1)

x1 x2 x3 x4 x5 x6 x7 x8 x9 x10 x11 x12 x13 x14 y
1 1  3  4  4  1  4  2  1  2  4   1   3   1   1   0
2 1  3  3  5  0  5  2  1  1  1   1   2   2   1   1
3 0  3  2  4  0  4  3  1  1  1   1   2   2   1   1
4 1  1  1  2  0  1  3  1  1  3   1   2   1   1   1
5 0  4  3  1  1  5  2  1  0  3   1   1   1   1   1
6 0  2  3  1  0  2  1  1  0  3   1   1   3   2   0

In [2]:
summary(data1)

       x1               x2              x3              x4       
 Min.   :0.0000   Min.   :1.000   Min.   :1.000   Min.   :1.000  
 1st Qu.:0.0000   1st Qu.:1.000   1st Qu.:1.000   1st Qu.:1.000  
 Median :1.0000   Median :2.000   Median :3.000   Median :2.000  
 Mean   :0.5306   Mean   :2.163   Mean   :2.531   Mean   :2.306  
 3rd Qu.:1.0000   3rd Qu.:3.000   3rd Qu.:4.000   3rd Qu.:3.000  
 Max.   :1.0000   Max.   :4.000   Max.   :4.000   Max.   :5.000  
       x5              x6              x7              x8       
 Min.   :0.000   Min.   :1.000   Min.   :1.000   Min.   :0.000  
 1st Qu.:0.000   1st Qu.:2.000   1st Qu.:2.000   1st Qu.:1.000  
 Median :1.000   Median :3.000   Median :2.000   Median :1.000  
 Mean   :0.551   Mean   :2.939   Mean   :2.224   Mean   :1.122  
 3rd Qu.:1.000   3rd Qu.:4.000   3rd Qu.:3.000   3rd Qu.:2.000  
 Max.   :1.000   Max.   :5.000   Max.   :4.000   Max.   :3.000  
       x9             x10             x11             x12       
 Min.   :0.000   M

在本题中，$y$是因变量，它只有两个值，所以可以把它看做成功概率为$P$的$Bernoulli$试验的结果。但是和单纯的$Bernoulli$试验不同，这里的概率$P$为$x_1$、$x_2$、$x_3$、$x_4$、$x_5$、$x_6$、$x_7$、$x_8$、$x_9$、$x_{10}$、$x_{11}$、$x_{12}$、$x_{13}$、$x_{14}$的函数。可以用下面的$Logistic$回归模型进行分析：

$ln(\frac{P}{1-P})=\beta_0+\beta_1x_1+\beta_2x_2+\beta_3x_3+\beta_4x_4+\beta_5x_5+\beta_6x_6+\beta_7x_7+\beta_8x_8+\beta_9x_9+\beta_{10}x_{10}+\beta_{11}x_{11}+\beta_{12}x_{12}+\beta_{13}x_{13}+\beta_{14}x_{14}+\epsilon$

In [3]:
logit.glm <- glm(y~.,family=binomial,data=data1)
summary(logit.glm)


Call:
glm(formula = y ~ ., family = binomial, data = data1)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)  6.84328    3.50393   1.953   0.0508 .
x1           0.37084    0.89132   0.416   0.6774  
x2          -1.24425    0.64367  -1.933   0.0532 .
x3          -1.28580    0.62028  -2.073   0.0382 *
x4           0.42060    0.40073   1.050   0.2939  
x5           0.53883    0.91688   0.588   0.5568  
x6           0.83929    0.54322   1.545   0.1223  
x7          -0.91736    0.69093  -1.328   0.1843  
x8          -0.61619    1.84859  -0.333   0.7389  
x9          -2.21468    2.07506  -1.067   0.2858  
x10          0.12711    0.32026   0.397   0.6914  
x11          0.08304    0.60020   0.138   0.8900  
x12          1.03718    2.09574   0.495   0.6207  
x13         -0.06666    0.68680  -0.097   0.9227  
x14         -0.43673    0.44680  -0.977   0.3283  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial famil

由此得到初步的$Logistic$回归模型：

$P=\frac{exp(6.84328+0.37084x_1-1.24425x_2-1.28580x_3+0.42060x_4+0.53883x_5+0.83929x_6-0.91736x_7-0.61619x_8-2.21468x_9+0.12711x_{10}+0.08304x_{11}+1.03718x_{12}-0.06666x_{13}-0.43673x_{14})}{1+exp(6.84328+0.37084x_1-1.24425x_2-1.28580x_3+0.42060x_4+0.53883x_5+0.83929x_6-0.91736x_7-0.61619x_8-2.21468x_9+0.12711x_{10}+0.08304x_{11}+1.03718x_{12}-0.06666x_{13}-0.43673x_{14})}$

即$logit(P)=6.84328+0.37084x_1-1.24425x_2-1.28580x_3+0.42060x_4+0.53883x_5+0.83929x_6-0.91736x_7-0.61619x_8-2.21468x_9+0.12711x_{10}+0.08304x_{11}+1.03718x_{12}-0.06666x_{13}-0.43673x_{14}$

在此模型中，由于参数$x_1$、$x_2$、$x_4$、$x_5$、$x_6$、$x_7$、$x_8$、$x_9$、$x_{10}$、$x_{11}$、$x_{12}$、$x_{13}$、$x_{14}$没有通过检验，可类似于线性模型，用$step( )$作变量筛选。

In [4]:
logit.step <- step(logit.glm,direction="both")

Start:  AIC=74.06
y ~ x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 + x10 + x11 + 
    x12 + x13 + x14

       Df Deviance    AIC
- x13   1   44.072 72.072
- x11   1   44.082 72.082
- x8    1   44.174 72.174
- x10   1   44.221 72.221
- x1    1   44.238 72.238
- x12   1   44.311 72.311
- x5    1   44.414 72.414
- x14   1   45.002 73.002
- x9    1   45.233 73.233
- x4    1   45.297 73.297
<none>      44.063 74.063
- x7    1   46.165 74.165
- x6    1   47.204 75.204
- x2    1   49.123 77.123
- x3    1   49.622 77.622

Step:  AIC=72.07
y ~ x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 + x10 + x11 + 
    x12 + x14

       Df Deviance    AIC
- x11   1   44.087 70.087
- x8    1   44.178 70.178
- x10   1   44.238 70.238
- x1    1   44.296 70.296
- x12   1   44.318 70.318
- x5    1   44.414 70.414
- x9    1   45.233 71.233
- x14   1   45.284 71.284
- x4    1   45.365 71.365
<none>      44.072 72.072
- x7    1   46.285 72.285
- x6    1   47.208 73.208
+ x13   1   44.063 74.063
- x2    1   49.163 75.16

In [5]:
summary(logit.step)


Call:
glm(formula = y ~ x2 + x3 + x6 + x7 + x9, family = binomial, 
    data = data1)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)   5.7176     2.3745   2.408   0.0160 *
x2           -0.5987     0.3949  -1.516   0.1294  
x3           -0.7847     0.4265  -1.840   0.0658 .
x6            0.5621     0.4054   1.387   0.1655  
x7           -0.7700     0.5501  -1.400   0.1616  
x9           -1.0577     0.5719  -1.849   0.0644 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 56.696  on 48  degrees of freedom
Residual deviance: 48.011  on 43  degrees of freedom
AIC: 60.011

Number of Fisher Scoring iterations: 5


可以看出新的回归方程为

$p=\frac{exp(5.7176-0.5987x_2-0.7847x_3+0.5621x_6-0.7700x_7-1.0577x_9)}{1+exp(5.7176-0.5987x_2-0.7847x_3+0.5621x_6-0.7700x_7-1.0577x_9)}$

可见，年龄、受教育程度、月收入、子女工作繁忙程度、女儿数量是探究老年人陪诊服务的使用意愿的关键因素。其中，年龄、受教育程度、子女工作繁忙程度、女儿数量这4个因素对老年人陪诊服务的使用意愿呈现一定的负相关性，即年龄越大、受教育程度越高、子女工作越繁忙、女儿数量越多的老年人使用陪诊服务的意愿越低；月收入这个因素对老年人陪诊服务的使用意愿呈现一定的正相关性，即月收入越高的老年人使用陪诊服务的意愿越高。